src/wp-includes/taxonomy.php

```diff
@@ -2825,6 +2825,22 @@ function wp_set_object_terms( $object_id, $terms, $taxonomy, $append = false ) {
 		$old_tt_ids = array();
 	}
 
+	/**
```

Comment:
I think this needs to be moved up by 15 lines above `if ( ! $append ) {`.

In line 2816, `$object_id` is already used, by altering `$object_id` in line 2842 this will normally produce `$old_tt_ids` that are no longer related to `$object_id`

Reply:
Fixed in [d382ac0](https://github.com/WordPress/wordpress-develop/pull/7660/commits/d382ac0019a3f395f0f74b990bf8893cb3c117b3)